lib/handleRetry.ts

```diff
@@ -1,10 +1,10 @@
 import { HTTPTimeout } from "./errors";
 
 export type RetryOptions = {
-  attempts?: number;
-  codes?: number[];
-  methods?: string[];
-  delay?: number;
+  attempts: number;
+  codes: number[];
+  methods: string[];
+  delay: number;
```

Comment:
User options use `Partial<RetryOptions>`, and we merge in defaults, so the source type should be non-optionally typed.